src/analyses/base.ml

```diff
@@ -2026,6 +2026,80 @@ struct
         M.warn ~category:(Behavior (Undefined InvalidMemoryDeallocation)) ~tags:[CWE 761] "Free of memory not at start of buffer in function %s for pointer %a" special_fn.vname d_exp ptr
     | _ -> M.warn ~category:MessageCategory.Analyzer "Pointer %a in function %s doesn't evaluate to a valid address." d_exp ptr special_fn.vname
 
+  (* Get the size of the smallest memory that dest points-to *)
+  let get_min_size_of_dest ctx dest =
+    let intdom_of_int x =
+      ID.of_int (Cilfacade.ptrdiff_ikind ()) (Z.of_int x)
+    in
+    let points_to_heap_only =
+      match ctx.ask (Queries.MayPointTo dest) with
+      | a when not (AD.is_top a) ->
+        AD.for_all (function
+            | Addr (v, _) -> ctx.ask (Queries.IsHeapVar v)
+            | _ -> false
+          ) a
+      | _ -> false
+    in
+    if points_to_heap_only then
```

Comment:
Why do I not need to use this for all of those things that are on the heap but only if all destinations are on the heap?

Reply:
Because `BlobSize` currently answers with `bot` if the points-to set does not contain heap variables only. Hence if the points-to set is mixed in the sense that it contains both heap and non-heap memory, we would get `bot` here. This is why we have this check here.

Alternatively, I guess we could skip the check and just join the result of `BlobSize` with the other sizes that we collect for the points-to set elements (i.e., the sizes, calculated in the `else` expression)?

Reply:
And maybe a second question: Why are you not working with the `to_index` of the offset in the Blob case here? I think something similar to what you did in the other PR should be possible, where for Blobs you get the length of the base address, add the numerical offset, and then check whether `offset + n <= size`?

Reply:
After the refactoring in the latest few commits, this should be fixed and we should be now able to account for offsets as well.